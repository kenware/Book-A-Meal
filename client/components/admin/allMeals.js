import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import PropTypes from 'prop-types';
import { ToastContainer, toast } from 'react-toastify';
import 'react-toastify/dist/ReactToastify.css';
import Modal from 'react-responsive-modal';
import Pagination from 'react-js-pagination';
import './index.scss';
import * as mealActions from '../../redux/Action/mealAction';


const limit = 6;
export class AllMeal extends Component {
  constructor(props) {
    super(props);
    this.state = {
      mealId: 0,
      mealName: '',
      mealImage: '',
      open: false,
      activePage: 1
    };
    this.cancelDelete = this.cancelDelete.bind(this);
    this.deleteMeal = this.deleteMeal.bind(this);
    this.handlePageChange = this.handlePageChange.bind(this);
    this.onOpenModal = this.onOpenModal.bind(this);
  }

  componentDidMount() {
    this.props.mealActions.getAllMeals(limit, 0);
  }

  /**
   * @param  {} mealId
   * @param  {} mealName
   * @param  {} mealImage
   * displays modal that give options to delete a meal or cancel delete
   */
  onOpenModal(mealId, mealName, mealImage) {
    this.setState({
      mealId, mealName, mealImage, open: true
    });
  }
  /**
   * closes modal without deleting a meal
   */
  cancelDelete() {
    this.setState({
      mealId: '', mealName: '', mealImage: '', open: false
    });
  }

  /**
   * calls a redux action
   * deletes a meal
   */
  deleteMeal() {
    if (this.state.mealId) {
      this.props.mealActions.deleteMeal(this.state.mealId);
      this.setState({
        mealId: '', mealName: '', mealImage: '', open: false
      });
    }
  }

  handlePageChange(pageNumber) {
    const offset = (pageNumber - 1) * limit;
    this.setState({ activePage: pageNumber });
    this.props.mealActions.getAllMeals(limit, offset);
  }

  render() {
    return (
      <div id="allmeal">
        <Modal open={this.state.open} onClose={this.cancelDelete} center>
          <br />
          <div className="modal-header">
            <p className="justify l-r-pad-text"> You are about to delete {this.state.mealName}</p>
          </div>
          <div className="modal-contents" style={{ justifyContent: 'center' }}>
            <img src={this.state.mealImage} className="rounded-circle img-height" alt="meal" />
          </div>
          <div className="modal-footer">
            <button className="remove-modal delete-meal" onClick={this.deleteMeal}>Delete</button>
          </div>
        </Modal>
        <h2>All Meal Options{this.props.successMessage.createMealSuccess }</h2>
        <h4 className="p-color text-center">Edit, delete or View all meal options</h4>
        <h2 className="danger">{this.props.successMessage.deleteMessage }</h2>
        {this.props.meals.rows.map(meal =>
        (
          <div className="contents container allMeal" key={meal.id}>
            <div className="content-wrap">
              <div className="col-meal l-r-pad-text">
                <img src={meal.image} className="rounded-circle img-height" alt="meals" />
              </div>
              <div className="col-meal">
                <h4 className="p-color"> Title</h4>
                <Link to={`/admin/edit/${meal.id}`} id="edit">{meal.name}</Link>
              </div>
              <div className="col-meal">
                <h4 className="p-color">Price</h4>{meal.price}
              </div>
              <div className="col-meal">
                <br />
                <Link to={`/admin/edit/${meal.id}`}><em className="fa fa-edit p-color" />&nbsp;Edit</Link><br /> <br />
                <span onClick={() => this.onOpenModal(meal.id, meal.name, meal.image)} className="p-color delete" id={meal.name} role="button"><em className="fa fa-trash p-color" />&nbsp;Delete</span>
              </div>
            </div>
          </div>
        ))}
        <div className="meal-pagination margin-bottom">
          {this.props.meals.rows.length > 0 ?
            <Pagination
              activePage={this.state.activePage}
              itemsCountPerPage={limit}
              totalItemsCount={Math.ceil(this.props.meals.count)}
              pageRangeDisplayed={4}
              onChange={this.handlePageChange}
            />
          :
            <span> Meal List Empty</span>}
        </div>
      </div>
    );
  }
}
AllMeal.propTypes = {
  meals: PropTypes.object.isRequired,
  successMessage: PropTypes.object.isRequired,
  mealActions: PropTypes.object.isRequired
};

export const mapStateToProps = state => ({
  meals: state.meals,
  successMessage: state.successMessage
});

export const mapDispatchToProps = dispatch => ({
  mealActions: bindActionCreators(mealActions, dispatch)
});

export default connect(mapStateToProps, mapDispatchToProps)(AllMeal);
