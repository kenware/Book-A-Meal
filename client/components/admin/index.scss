.form-label {
    flex:2;
    padding: 10px 10px 10px 10px;
    margin-right: 10px;
}
.detail-contents{
    display:flex;
    justify-content: space-around;
  }
  .detail-img{
    width:60%;
    height: 60%;
  }
  .detail-text{
   margin-top:1rem;
  }
  @media screen and (max-width: 600px) {
  .largeScreen-header{
     display: none;
    }
  }
  @media screen and (min-width: 600px) {
    .smallScreen-header{
       display: none;
      }
    }
    // react paginate components
    #react-paginate ul {
      display: inline-block;
      padding-left: 15px;
      padding-right: 15px;
  }
  
  #react-paginate li {
      display: inline-block;
  }
  
  #react-paginate .break a {
      cursor: default;
  }
