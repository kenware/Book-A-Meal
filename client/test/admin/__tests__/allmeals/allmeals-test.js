import React from 'react';
import { shallow } from 'enzyme';
import { AllMeal, mapDispatchToProps, mapStateToProps } from '../../../../components/admin/allMeals';
import { props, emptyProps } from '../../adminMockData';

describe('Test Component of ALLMeals component', () => {
  it('renders correctly', () => {
    const tree = shallow(<AllMeal {...props} />);
    expect(tree).toMatchSnapshot();
  });

  it('should display two meals contents', () => {
    const tree = shallow(<AllMeal {...props} />);
    expect(tree.find('.contents')).toHaveLength(2);
  });

  it('should display 1 modal-order contents', () => {
    const tree = shallow(<AllMeal {...props} />);
    expect(tree.find('.modal-order')).toHaveLength(1);
  });

  it('should display 3 modal-order-content class contents', () => {
    const tree = shallow(<AllMeal {...props} />);
    expect(tree.find('.modal-order-content')).toHaveLength(3);
  });

  it('should display 8 col-meal class from the meals contents', () => {
    const tree = shallow(<AllMeal {...props} />);
    expect(tree.find('.col-meal')).toHaveLength(8);
  });

  it('should have 17 divs', () => {
    const tree = shallow(<AllMeal {...props} />);
    expect(tree.find('div')).toHaveLength(17);
  });

  it('should display 2 img tag from the meals contents', () => {
    const tree = shallow(<AllMeal {...props} />);
    expect(tree.find('img')).toHaveLength(3);
  });

  it('should display no meal meals with empty meal props', () => {
    const tree = shallow(<AllMeal {...emptyProps} />);
    expect(tree.find('.contents')).toHaveLength(0);
  });

  it('should display no menu meals with empty meal props', () => {
    const tree = shallow(<AllMeal {...props} />);
    expect(tree.find('Link')).toHaveLength(4);
  });

  it('should respond to a click event by dispalying a modal when caterer want to delete a meal by selecting from meal options', () => {
    const tree = shallow(<AllMeal {...props} />);
    // onClick event
    tree.find('#rice').simulate('click');
    expect(tree).toMatchSnapshot();
  });
  it('should respond to deleteMeal method', () => {
    const tree = shallow(<AllMeal {...props} />);
    // onClick event
    const wrapper = tree.instance();
    wrapper.deleteMeal();
    expect(tree).toMatchSnapshot();
  });
  it('should respond to deleteMeal method ', () => {
    const tree = shallow(<AllMeal {...props} />);
    // onClick event
    const wrapper = tree.instance();
    wrapper.cancelDelete();
    expect(tree).toMatchSnapshot();
  });
  it('should call deletMeal method onclick event', () => {
    const deleteMeal = jest.spyOn(AllMeal.prototype, 'deleteMeal');
    const tree = shallow(<AllMeal {...props} />);
    // onClick event
    tree.find('#rice').simulate('click');
    tree.find('.delete-meal').simulate('click');
    expect(tree).toMatchSnapshot();
    expect(deleteMeal).toHaveBeenCalled();
  });

  it('should cancel/remove modal that delete meal using cancelAdd method onclick event', () => {
    const cancelDelete = jest.spyOn(AllMeal.prototype, 'cancelDelete');
    const tree = shallow(<AllMeal {...props} />);
    // onClick event
    tree.find('#rice').simulate('click');
    tree.find('.cancel-delete').simulate('click');
    expect(tree).toMatchSnapshot();
    expect(cancelDelete).toHaveBeenCalled();
  });
  it('should respond to mapStateToProps methods', () => {
    const ownProps = { match: { params: { mealId: 1 } } };
    const tree = mapStateToProps(props, ownProps);
    expect(tree).toMatchSnapshot();
  });
  it('should respond to mapDispatchToProps methods', () => {
    const tree = mapDispatchToProps(emptyProps);
    expect(tree).toMatchSnapshot();
  });
});
