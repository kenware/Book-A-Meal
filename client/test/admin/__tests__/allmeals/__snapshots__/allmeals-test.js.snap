// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Component of ALLMeals component renders correctly 1`] = `
<div>
  <div
    className="modal-order modal"
  >
    <button
      className="remove-modal"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "red",
          "display": "block",
          "float": "right",
        }
      }
    >
      ×
    </button>
    <div
      className="modal-order-content"
    >
      <img
        alt="meal"
        className="rounded-circle img-height"
        src=""
      />
    </div>
    <div
      className="modal-order-content"
    >
      <p
        className="justify l-r-pad-text"
      >
         You are about to delete 
      </p>
    </div>
    <div
      className="modal-order-content"
    >
      <button
        className="remove-modal delete-meal"
        onClick={[Function]}
      >
        Delete
      </button>
      <button
        className="remove-modal cancel-delete"
        onClick={[Function]}
      >
        Cancel
      </button>
    </div>
  </div>
  <h2>
    All Meal Options
    meal created
  </h2>
  <h4
    className="p-color text-center"
  >
    Edit, delete or View all meal options
  </h4>
  <h4
    className="danger text-center"
  />
  <div
    className="contents container"
    key="1"
  >
    <div
      className="content-wrap"
    >
      <div
        className="col-meal l-r-pad-text"
      >
        <img
          alt="meals"
          className="rounded-circle img-height"
        />
      </div>
      <div
        className="col-meal"
      >
        <h4
          className="p-color"
        >
           Title
        </h4>
        <Link
          replace={false}
          to="/admin/edit/1"
        >
          rice
        </Link>
      </div>
      <div
        className="col-meal"
      >
        <h4
          className="p-color"
        >
          Price
        </h4>
        200
      </div>
      <div
        className="col-meal"
      >
        <br />
        <Link
          replace={false}
          to="/admin/edit/1"
        >
          <em
            className="fa fa-edit p-color"
          />
           Edit
        </Link>
        <br />
         
        <br />
        <span
          className="p-color"
          id="rice"
          onClick={[Function]}
          role="button"
        >
          <em
            className="fa fa-trash p-color"
          />
           Delete
        </span>
      </div>
    </div>
  </div>
  <div
    className="contents container"
    key="2"
  >
    <div
      className="content-wrap"
    >
      <div
        className="col-meal l-r-pad-text"
      >
        <img
          alt="meals"
          className="rounded-circle img-height"
        />
      </div>
      <div
        className="col-meal"
      >
        <h4
          className="p-color"
        >
           Title
        </h4>
        <Link
          replace={false}
          to="/admin/edit/2"
        >
          beans
        </Link>
      </div>
      <div
        className="col-meal"
      >
        <h4
          className="p-color"
        >
          Price
        </h4>
        200
      </div>
      <div
        className="col-meal"
      >
        <br />
        <Link
          replace={false}
          to="/admin/edit/2"
        >
          <em
            className="fa fa-edit p-color"
          />
           Edit
        </Link>
        <br />
         
        <br />
        <span
          className="p-color"
          id="beans"
          onClick={[Function]}
          role="button"
        >
          <em
            className="fa fa-trash p-color"
          />
           Delete
        </span>
      </div>
    </div>
  </div>
  <div
    className="pagination"
  >
    <div
      className="prev p-color"
    >
         
      <button
        onClick={[Function]}
      >
         
        <em
          className="fa fa-angle-left"
        />
         PREV 
      </button>
    </div>
    <div
      className="current p-color"
    >
      <button>
        1
         
      </button>
    </div>
    <div
      className="next p-color"
    >
      <button
        onClick={[Function]}
      >
        NEXT 
        <em
          className="fa fa-angle-right"
        />
      </button>
        
       
    </div>
  </div>
</div>
`;

exports[`Test Component of ALLMeals component should call deletMeal method onclick event 1`] = `
<div>
  <div
    className="modal-order modal"
  >
    <button
      className="remove-modal"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "red",
          "display": "block",
          "float": "right",
        }
      }
    >
      ×
    </button>
    <div
      className="modal-order-content"
    >
      <img
        alt="meal"
        className="rounded-circle img-height"
        src=""
      />
    </div>
    <div
      className="modal-order-content"
    >
      <p
        className="justify l-r-pad-text"
      >
         You are about to delete 
      </p>
    </div>
    <div
      className="modal-order-content"
    >
      <button
        className="remove-modal delete-meal"
        onClick={[Function]}
      >
        Delete
      </button>
      <button
        className="remove-modal cancel-delete"
        onClick={[Function]}
      >
        Cancel
      </button>
    </div>
  </div>
  <h2>
    All Meal Options
    meal created
  </h2>
  <h4
    className="p-color text-center"
  >
    Edit, delete or View all meal options
  </h4>
  <h4
    className="danger text-center"
  />
  <div
    className="contents container"
    key="1"
  >
    <div
      className="content-wrap"
    >
      <div
        className="col-meal l-r-pad-text"
      >
        <img
          alt="meals"
          className="rounded-circle img-height"
        />
      </div>
      <div
        className="col-meal"
      >
        <h4
          className="p-color"
        >
           Title
        </h4>
        <Link
          replace={false}
          to="/admin/edit/1"
        >
          rice
        </Link>
      </div>
      <div
        className="col-meal"
      >
        <h4
          className="p-color"
        >
          Price
        </h4>
        200
      </div>
      <div
        className="col-meal"
      >
        <br />
        <Link
          replace={false}
          to="/admin/edit/1"
        >
          <em
            className="fa fa-edit p-color"
          />
           Edit
        </Link>
        <br />
         
        <br />
        <span
          className="p-color"
          id="rice"
          onClick={[Function]}
          role="button"
        >
          <em
            className="fa fa-trash p-color"
          />
           Delete
        </span>
      </div>
    </div>
  </div>
  <div
    className="contents container"
    key="2"
  >
    <div
      className="content-wrap"
    >
      <div
        className="col-meal l-r-pad-text"
      >
        <img
          alt="meals"
          className="rounded-circle img-height"
        />
      </div>
      <div
        className="col-meal"
      >
        <h4
          className="p-color"
        >
           Title
        </h4>
        <Link
          replace={false}
          to="/admin/edit/2"
        >
          beans
        </Link>
      </div>
      <div
        className="col-meal"
      >
        <h4
          className="p-color"
        >
          Price
        </h4>
        200
      </div>
      <div
        className="col-meal"
      >
        <br />
        <Link
          replace={false}
          to="/admin/edit/2"
        >
          <em
            className="fa fa-edit p-color"
          />
           Edit
        </Link>
        <br />
         
        <br />
        <span
          className="p-color"
          id="beans"
          onClick={[Function]}
          role="button"
        >
          <em
            className="fa fa-trash p-color"
          />
           Delete
        </span>
      </div>
    </div>
  </div>
  <div
    className="pagination"
  >
    <div
      className="prev p-color"
    >
         
      <button
        onClick={[Function]}
      >
         
        <em
          className="fa fa-angle-left"
        />
         PREV 
      </button>
    </div>
    <div
      className="current p-color"
    >
      <button>
        1
         
      </button>
    </div>
    <div
      className="next p-color"
    >
      <button
        onClick={[Function]}
      >
        NEXT 
        <em
          className="fa fa-angle-right"
        />
      </button>
        
       
    </div>
  </div>
</div>
`;

exports[`Test Component of ALLMeals component should cancel/remove modal that delete meal using cancelAdd method onclick event 1`] = `
<div>
  <div
    className="modal-order modal"
  >
    <button
      className="remove-modal"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "red",
          "display": "block",
          "float": "right",
        }
      }
    >
      ×
    </button>
    <div
      className="modal-order-content"
    >
      <img
        alt="meal"
        className="rounded-circle img-height"
        src=""
      />
    </div>
    <div
      className="modal-order-content"
    >
      <p
        className="justify l-r-pad-text"
      >
         You are about to delete 
      </p>
    </div>
    <div
      className="modal-order-content"
    >
      <button
        className="remove-modal delete-meal"
        onClick={[Function]}
      >
        Delete
      </button>
      <button
        className="remove-modal cancel-delete"
        onClick={[Function]}
      >
        Cancel
      </button>
    </div>
  </div>
  <h2>
    All Meal Options
    meal created
  </h2>
  <h4
    className="p-color text-center"
  >
    Edit, delete or View all meal options
  </h4>
  <h4
    className="danger text-center"
  />
  <div
    className="contents container"
    key="1"
  >
    <div
      className="content-wrap"
    >
      <div
        className="col-meal l-r-pad-text"
      >
        <img
          alt="meals"
          className="rounded-circle img-height"
        />
      </div>
      <div
        className="col-meal"
      >
        <h4
          className="p-color"
        >
           Title
        </h4>
        <Link
          replace={false}
          to="/admin/edit/1"
        >
          rice
        </Link>
      </div>
      <div
        className="col-meal"
      >
        <h4
          className="p-color"
        >
          Price
        </h4>
        200
      </div>
      <div
        className="col-meal"
      >
        <br />
        <Link
          replace={false}
          to="/admin/edit/1"
        >
          <em
            className="fa fa-edit p-color"
          />
           Edit
        </Link>
        <br />
         
        <br />
        <span
          className="p-color"
          id="rice"
          onClick={[Function]}
          role="button"
        >
          <em
            className="fa fa-trash p-color"
          />
           Delete
        </span>
      </div>
    </div>
  </div>
  <div
    className="contents container"
    key="2"
  >
    <div
      className="content-wrap"
    >
      <div
        className="col-meal l-r-pad-text"
      >
        <img
          alt="meals"
          className="rounded-circle img-height"
        />
      </div>
      <div
        className="col-meal"
      >
        <h4
          className="p-color"
        >
           Title
        </h4>
        <Link
          replace={false}
          to="/admin/edit/2"
        >
          beans
        </Link>
      </div>
      <div
        className="col-meal"
      >
        <h4
          className="p-color"
        >
          Price
        </h4>
        200
      </div>
      <div
        className="col-meal"
      >
        <br />
        <Link
          replace={false}
          to="/admin/edit/2"
        >
          <em
            className="fa fa-edit p-color"
          />
           Edit
        </Link>
        <br />
         
        <br />
        <span
          className="p-color"
          id="beans"
          onClick={[Function]}
          role="button"
        >
          <em
            className="fa fa-trash p-color"
          />
           Delete
        </span>
      </div>
    </div>
  </div>
  <div
    className="pagination"
  >
    <div
      className="prev p-color"
    >
         
      <button
        onClick={[Function]}
      >
         
        <em
          className="fa fa-angle-left"
        />
         PREV 
      </button>
    </div>
    <div
      className="current p-color"
    >
      <button>
        1
         
      </button>
    </div>
    <div
      className="next p-color"
    >
      <button
        onClick={[Function]}
      >
        NEXT 
        <em
          className="fa fa-angle-right"
        />
      </button>
        
       
    </div>
  </div>
</div>
`;

exports[`Test Component of ALLMeals component should respond to a click event by dispalying a modal when caterer want to delete a meal by selecting from meal options 1`] = `
<div>
  <div
    className="modal-order "
  >
    <button
      className="remove-modal"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "red",
          "display": "block",
          "float": "right",
        }
      }
    >
      ×
    </button>
    <div
      className="modal-order-content"
    >
      <img
        alt="meal"
        className="rounded-circle img-height"
      />
    </div>
    <div
      className="modal-order-content"
    >
      <p
        className="justify l-r-pad-text"
      >
         You are about to delete 
        rice
      </p>
    </div>
    <div
      className="modal-order-content"
    >
      <button
        className="remove-modal delete-meal"
        onClick={[Function]}
      >
        Delete
      </button>
      <button
        className="remove-modal cancel-delete"
        onClick={[Function]}
      >
        Cancel
      </button>
    </div>
  </div>
  <h2>
    All Meal Options
    meal created
  </h2>
  <h4
    className="p-color text-center"
  >
    Edit, delete or View all meal options
  </h4>
  <h4
    className="danger text-center"
  />
  <div
    className="contents container"
    key="1"
  >
    <div
      className="content-wrap"
    >
      <div
        className="col-meal l-r-pad-text"
      >
        <img
          alt="meals"
          className="rounded-circle img-height"
        />
      </div>
      <div
        className="col-meal"
      >
        <h4
          className="p-color"
        >
           Title
        </h4>
        <Link
          replace={false}
          to="/admin/edit/1"
        >
          rice
        </Link>
      </div>
      <div
        className="col-meal"
      >
        <h4
          className="p-color"
        >
          Price
        </h4>
        200
      </div>
      <div
        className="col-meal"
      >
        <br />
        <Link
          replace={false}
          to="/admin/edit/1"
        >
          <em
            className="fa fa-edit p-color"
          />
           Edit
        </Link>
        <br />
         
        <br />
        <span
          className="p-color"
          id="rice"
          onClick={[Function]}
          role="button"
        >
          <em
            className="fa fa-trash p-color"
          />
           Delete
        </span>
      </div>
    </div>
  </div>
  <div
    className="contents container"
    key="2"
  >
    <div
      className="content-wrap"
    >
      <div
        className="col-meal l-r-pad-text"
      >
        <img
          alt="meals"
          className="rounded-circle img-height"
        />
      </div>
      <div
        className="col-meal"
      >
        <h4
          className="p-color"
        >
           Title
        </h4>
        <Link
          replace={false}
          to="/admin/edit/2"
        >
          beans
        </Link>
      </div>
      <div
        className="col-meal"
      >
        <h4
          className="p-color"
        >
          Price
        </h4>
        200
      </div>
      <div
        className="col-meal"
      >
        <br />
        <Link
          replace={false}
          to="/admin/edit/2"
        >
          <em
            className="fa fa-edit p-color"
          />
           Edit
        </Link>
        <br />
         
        <br />
        <span
          className="p-color"
          id="beans"
          onClick={[Function]}
          role="button"
        >
          <em
            className="fa fa-trash p-color"
          />
           Delete
        </span>
      </div>
    </div>
  </div>
  <div
    className="pagination"
  >
    <div
      className="prev p-color"
    >
         
      <button
        onClick={[Function]}
      >
         
        <em
          className="fa fa-angle-left"
        />
         PREV 
      </button>
    </div>
    <div
      className="current p-color"
    >
      <button>
        1
         
      </button>
    </div>
    <div
      className="next p-color"
    >
      <button
        onClick={[Function]}
      >
        NEXT 
        <em
          className="fa fa-angle-right"
        />
      </button>
        
       
    </div>
  </div>
</div>
`;

exports[`Test Component of ALLMeals component should respond to deleteMeal method  1`] = `
<div>
  <div
    className="modal-order modal"
  >
    <button
      className="remove-modal"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "red",
          "display": "block",
          "float": "right",
        }
      }
    >
      ×
    </button>
    <div
      className="modal-order-content"
    >
      <img
        alt="meal"
        className="rounded-circle img-height"
        src=""
      />
    </div>
    <div
      className="modal-order-content"
    >
      <p
        className="justify l-r-pad-text"
      >
         You are about to delete 
      </p>
    </div>
    <div
      className="modal-order-content"
    >
      <button
        className="remove-modal delete-meal"
        onClick={[Function]}
      >
        Delete
      </button>
      <button
        className="remove-modal cancel-delete"
        onClick={[Function]}
      >
        Cancel
      </button>
    </div>
  </div>
  <h2>
    All Meal Options
    meal created
  </h2>
  <h4
    className="p-color text-center"
  >
    Edit, delete or View all meal options
  </h4>
  <h4
    className="danger text-center"
  />
  <div
    className="contents container"
    key="1"
  >
    <div
      className="content-wrap"
    >
      <div
        className="col-meal l-r-pad-text"
      >
        <img
          alt="meals"
          className="rounded-circle img-height"
        />
      </div>
      <div
        className="col-meal"
      >
        <h4
          className="p-color"
        >
           Title
        </h4>
        <Link
          replace={false}
          to="/admin/edit/1"
        >
          rice
        </Link>
      </div>
      <div
        className="col-meal"
      >
        <h4
          className="p-color"
        >
          Price
        </h4>
        200
      </div>
      <div
        className="col-meal"
      >
        <br />
        <Link
          replace={false}
          to="/admin/edit/1"
        >
          <em
            className="fa fa-edit p-color"
          />
           Edit
        </Link>
        <br />
         
        <br />
        <span
          className="p-color"
          id="rice"
          onClick={[Function]}
          role="button"
        >
          <em
            className="fa fa-trash p-color"
          />
           Delete
        </span>
      </div>
    </div>
  </div>
  <div
    className="contents container"
    key="2"
  >
    <div
      className="content-wrap"
    >
      <div
        className="col-meal l-r-pad-text"
      >
        <img
          alt="meals"
          className="rounded-circle img-height"
        />
      </div>
      <div
        className="col-meal"
      >
        <h4
          className="p-color"
        >
           Title
        </h4>
        <Link
          replace={false}
          to="/admin/edit/2"
        >
          beans
        </Link>
      </div>
      <div
        className="col-meal"
      >
        <h4
          className="p-color"
        >
          Price
        </h4>
        200
      </div>
      <div
        className="col-meal"
      >
        <br />
        <Link
          replace={false}
          to="/admin/edit/2"
        >
          <em
            className="fa fa-edit p-color"
          />
           Edit
        </Link>
        <br />
         
        <br />
        <span
          className="p-color"
          id="beans"
          onClick={[Function]}
          role="button"
        >
          <em
            className="fa fa-trash p-color"
          />
           Delete
        </span>
      </div>
    </div>
  </div>
  <div
    className="pagination"
  >
    <div
      className="prev p-color"
    >
         
      <button
        onClick={[Function]}
      >
         
        <em
          className="fa fa-angle-left"
        />
         PREV 
      </button>
    </div>
    <div
      className="current p-color"
    >
      <button>
        1
         
      </button>
    </div>
    <div
      className="next p-color"
    >
      <button
        onClick={[Function]}
      >
        NEXT 
        <em
          className="fa fa-angle-right"
        />
      </button>
        
       
    </div>
  </div>
</div>
`;

exports[`Test Component of ALLMeals component should respond to deleteMeal method 1`] = `
<div>
  <div
    className="modal-order modal"
  >
    <button
      className="remove-modal"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "red",
          "display": "block",
          "float": "right",
        }
      }
    >
      ×
    </button>
    <div
      className="modal-order-content"
    >
      <img
        alt="meal"
        className="rounded-circle img-height"
        src=""
      />
    </div>
    <div
      className="modal-order-content"
    >
      <p
        className="justify l-r-pad-text"
      >
         You are about to delete 
      </p>
    </div>
    <div
      className="modal-order-content"
    >
      <button
        className="remove-modal delete-meal"
        onClick={[Function]}
      >
        Delete
      </button>
      <button
        className="remove-modal cancel-delete"
        onClick={[Function]}
      >
        Cancel
      </button>
    </div>
  </div>
  <h2>
    All Meal Options
    meal created
  </h2>
  <h4
    className="p-color text-center"
  >
    Edit, delete or View all meal options
  </h4>
  <h4
    className="danger text-center"
  />
  <div
    className="contents container"
    key="1"
  >
    <div
      className="content-wrap"
    >
      <div
        className="col-meal l-r-pad-text"
      >
        <img
          alt="meals"
          className="rounded-circle img-height"
        />
      </div>
      <div
        className="col-meal"
      >
        <h4
          className="p-color"
        >
           Title
        </h4>
        <Link
          replace={false}
          to="/admin/edit/1"
        >
          rice
        </Link>
      </div>
      <div
        className="col-meal"
      >
        <h4
          className="p-color"
        >
          Price
        </h4>
        200
      </div>
      <div
        className="col-meal"
      >
        <br />
        <Link
          replace={false}
          to="/admin/edit/1"
        >
          <em
            className="fa fa-edit p-color"
          />
           Edit
        </Link>
        <br />
         
        <br />
        <span
          className="p-color"
          id="rice"
          onClick={[Function]}
          role="button"
        >
          <em
            className="fa fa-trash p-color"
          />
           Delete
        </span>
      </div>
    </div>
  </div>
  <div
    className="contents container"
    key="2"
  >
    <div
      className="content-wrap"
    >
      <div
        className="col-meal l-r-pad-text"
      >
        <img
          alt="meals"
          className="rounded-circle img-height"
        />
      </div>
      <div
        className="col-meal"
      >
        <h4
          className="p-color"
        >
           Title
        </h4>
        <Link
          replace={false}
          to="/admin/edit/2"
        >
          beans
        </Link>
      </div>
      <div
        className="col-meal"
      >
        <h4
          className="p-color"
        >
          Price
        </h4>
        200
      </div>
      <div
        className="col-meal"
      >
        <br />
        <Link
          replace={false}
          to="/admin/edit/2"
        >
          <em
            className="fa fa-edit p-color"
          />
           Edit
        </Link>
        <br />
         
        <br />
        <span
          className="p-color"
          id="beans"
          onClick={[Function]}
          role="button"
        >
          <em
            className="fa fa-trash p-color"
          />
           Delete
        </span>
      </div>
    </div>
  </div>
  <div
    className="pagination"
  >
    <div
      className="prev p-color"
    >
         
      <button
        onClick={[Function]}
      >
         
        <em
          className="fa fa-angle-left"
        />
         PREV 
      </button>
    </div>
    <div
      className="current p-color"
    >
      <button>
        1
         
      </button>
    </div>
    <div
      className="next p-color"
    >
      <button
        onClick={[Function]}
      >
        NEXT 
        <em
          className="fa fa-angle-right"
        />
      </button>
        
       
    </div>
  </div>
</div>
`;

exports[`Test Component of ALLMeals component should respond to mapDispatchToProps methods 1`] = `
Object {
  "mealActions": Object {
    "createMeal": [Function],
    "deleteMeal": [Function],
    "getAllMeals": [Function],
    "loadAllMeals": [Function],
    "loadErrorMessage": [Function],
    "loadMostOrdered": [Function],
    "loadMostOrderedMeal": [Function],
    "loadSuccessMessage": [Function],
    "updateMeal": [Function],
  },
}
`;

exports[`Test Component of ALLMeals component should respond to mapStateToProps methods 1`] = `
Object {
  "meals": Object {
    "rows": Array [
      Object {
        "descrption": "Very delicious",
        "id": 1,
        "name": "rice",
        "price": "200",
      },
      Object {
        "descrption": "Very delicious",
        "id": 2,
        "name": "beans",
        "price": "200",
      },
    ],
  },
  "successMessage": Object {
    "createMealSuccess": "meal created",
  },
}
`;
